### DISCUSSION TITLE
E(rase) command should always work

### DISCUSSION THREAD

**Discussion by prlaba** [TomNisbet/TommyPROM#51] [Ideas]
When I attempt to use the E(rase) command with certain EEPROMs, TommyPROM returns the message NOT SUPPORTED.

I assume this is because only certain EEPROMs have an Erase command.

But from a user's perspective, they probably don't care how the EEPROM gets erased.

Instead of returning NOT SUPPORTED, TommyPROM could go ahead and fill each cell w the erase value 0xff.  Since that could take longer than using a chip's built-in Erase command, TommyPROM could do something like:

>E
This chip does not support ERASE
Using FILL to erase...

Or it could even prompt the user:

>E
This chip does not support ERASE. 
Would you like to use FILL instead (Y/N)?

**Comment by TomNisbet** [comment]
The Erase command was added recently to support some older flash chips that need to be explicitly erased before programming.  More modern chips, like the 28C256 and the SST39F chips, erase themselves on a new write operation, so there is no need to ever erase them.  If you want a chip to be all FF for some reason, you can always use the Fill command, but there's usually no need to do this.

**Reply by prlaba** [reply]
Thanks Tom for clarifying why the command is supported for some chips but not others.

But I think the E command is a useful and quick way to put any EPROM into a known state.  It would be easy enough for the TommyPROM code to perform the fill operation itself.

I'm envisioning the TommyPROM as being more of a general EEPROM programmer than an experimenter's programming device.  That's the reason for some of my suggestions.  Probably not what you had in mind.
